### DISCUSSION TITLE
No Audio Output Over USB in Build 20221117

### DISCUSSION THREAD

**Discussion by digixmax** [rern/rAudio#1173] [Bug Reports]
I lost audio signal over USB following a fresh flash of build 20221117 and a subsequent restoration of backup configuration used in previous build 20220814.

This happened with both my RPi-3B and RPi-4, and both when using a USB audio dongle and (alternatively) a USB DAC (SMSL M500).  The player correctly identified the USB audio devices, but there is no audio signal and hence sound. I could get the audio signal to the onboard headphone jack, so there is no issue with the audio renderer (Spotify).

The issue went away when I went back to build 20220904 with the same exact backup configuration.

**Comment by kw-sudo** [comment]
Hi,

similar issue [here](https://github.com/rern/rAudio-1/discussions/1066) (ongoing over a couple of versions). My only way to 'fix' it is to switch between the audio outputs (onboard headphones <=> amp).

But I was able to narrow down the error. It only happens after a reboot ('cold and warm start')

**Comment by hjfisch** [comment]
Hello ,
I'd several issues too after restoring my backup . I did a clean install and updated to the newest Version afterwards.
New configuring was a bit of pain , but now I've no more probs.Also had probs with booting from sd-card with extension cable , that's why I'm booting from USB-Stick now. But like booting from sd-card , not all USB-Sticks will work with the Raspberry. Best working Stick for me is "Verbatim Store 'n' Stay Nano USB-Stick, USB-3.2 Gen1, 16 GB".

**Reply by digixmax** [reply]
Thanks, I had also thought of "restoring and/or using a backup config" being a potential cause (e.g., due to lack of backward compatibility in configuration settings between new and old releases -- I learned to be frequently the case in home-router firmware upgrade), but I have dreaded to do a clean install of 20221117 as I would be hard pressed to recall all the configuration settings that need to be re-entered.

**Comment by R101G** [comment]
I noticed a while ago that during boot the Pi USB port does not power up cleanly. The 5v comes on, then off for a moment, and back on again. I had never noticed this happening before, so assumed a kernel update was responsible. Could it be that the DACs power-up sequence is being corrupted by this interruption?

**Reply by kw-sudo** [reply]
The strange thing, the DAC shows up in r/Audio but it doesn't play. It only works when I change the output source, as mentioned above.

PS: since my RPi runs 24/7, this only happens with updates or similar
PPS: It also doesn't help to define the amp as the output source in mpd.conf.

[edit]:

In fact, I can confirm what I just wrote. After a major update I restarted RPi and my Nubert Amp/DAC shows up. Then I have to switch once to the headphones and back again, after that 'a sound comes out'.


![A1BB2373-6EA8-4E90-997C-827DD90442FF](https://user-images.githubusercontent.com/104136223/211488385-81c57c5f-d7fa-498b-bf76-418dfd128cf9.jpeg)

**Comment by R101G** [comment]
Try disabling on-board audio _(dtparam=audio=off_ in /boot/config.txt). If no sound comes out, but unplugging and plugging in the DAC restores it, this would imply that the DAC is not initialising properly due to the messy USB power-up.

**Reply by kw-sudo** [reply]
> If no sound comes out, but unplugging and plugging in the DAC restores it, 

Exact this happens...

>this would imply that the DAC is not initialising properly due to the messy USB power-up.

My amp has an "auto power function". Strange, that the amp recognizes the upcoming RPi boot sequence, turns on, but finally no sound. 

I don't need that headphone output, can I disable it? In _alsamixer_ I set the amp (card1) to default (F6), but no change

PS: asound.conf

`defaults.pcm.card 1
defaults.ctl.card 1`

**Reply by digixmax** [reply]
I tried swapping between my USB DAC and USB audio dongle by removing one and inserting the other one -- rAudio correctly identified the newly inserted USB device every time but still no sound.

I tried switching output between USB device and on-board audio jack and back -- no go either on getting sound out to USB device.

Everything works fine again, no issue whatsoever, after I reverted back to build # 20220904 -- so it seems to me that some bug crept in with the build # 20221117.

**Reply by kw-sudo** [reply]
...I forgot to mention that my amp has a built-in DAC, if that makes a difference. The 'audio chain' is:

`NAS --> RJ45 connection--> RasPI@r/Audio --> USB A-B --> amp/DAC USB input`

**Comment by R101G** [comment]
Dealing with two different problems in the same thread is not a good idea. I suggest that @master-kw  starts a new discussion. It might help if everyone updated to the latest rAudio version and kernel, too.

**Reply by kw-sudo** [reply]
> @master-kw starts a new discussion

As already written above, the [topic](https://github.com/rern/rAudio-1/discussions/1066) already open. It sounded similar for me, so I continued discussion here. :-) 

PS: I update regularly, most recently this morning

**Reply by digixmax** [reply]
Did any of your most recent updates to 20221117 resolve (i.e., without any dependence on some quirky workaround) these USB output related issues?

**Reply by kw-sudo** [reply]
No, it didn’t/doesn’t. The setup is quite the same. 

As I already said, it only happens after a reboot. Basically, RPi runs 24/7, amp is standby (if not in use). If I turn on the amp and start playing music, everything’s fine…until a reboot (e.g. after Linux update).

**Reply by cmh714** [reply]
The Amp needs to be powered up BEFORE the pi comes up. Thats been the case since the beginning and for every music player that I have ever used.

**Reply by kw-sudo** [reply]
@cmh714: I've just (re-)tested it, it doesn't matter. 

config Amp:
- power on
- auto power off
- USB input active

...reboot RPi, same issue, no sound --> switch sound cards, it works. Basically, it's not tragic, since I don't restart the Raspberry "twenty times" a day. ;-)

**Reply by cmh714** [reply]
Hmm, not sure what your issue is. I am working fine with a similar setup. I flashed clean with this version a few weeks ago. Fully updated both rAudio and pacman.....I occasionally will have an issue that I generally chalk up to some "timing" thing :) and usually a reboot fixes it.

GL!

**Reply by digixmax** [reply]
I can confirm that there is no dependency on the ordering of powering-up devices when running build # 20220904.  In the absence of external USB device rAudio on my RPi-4 upon boot-up defaults to the on-board audio output, but automatically switches to USB output as soon as my USB DAC gets powered up.

**Comment by R101G** [comment]
So far as I can see, @digixmax  gets sound after unplugging and plugging the USB connection, and @master-kw does not, so probably not quite the same problem?

Either way, it does not seem reasonable to submit a bug report unless it applies to the current version. Have you actually tried updating  rAudio and the kernel to the latest versions available? Is the problem still present if you do?

**Reply by digixmax** [reply]
> So far as I can see, @digixmax gets sound after unplugging and plugging the USB connection, and @master-kw does not, so probably not quite the same problem?
> 
It's the other way around, I got no sound regardless of any attempt of unplugging/replugging or alternate powering-up sequencing.

I will probably attempt a clean flash + update, then fresh configuration from scratch to see if that would work.

**Reply by kw-sudo** [reply]
@R101G: Yes, sure, I agreed above.

>If no sound comes out, but unplugging and plugging in the DAC restores it,

> >Exact this happens...

I have the choice between unplugging or switching cards (@r/Audio player settings). Both succeeds

**Comment by digixmax** [comment]
Ok - I did a clean flash of 20221117, upgraded to 20230101, and then re-entered my configuration settings from scratch.

[Note: I attempted to restore backup configuration from used since previous build 20220814 -- the restore operation appeared successfully completed but rAudio configuration settings remained unchanged following reboot!]

Sadly, I still get no sound from my USB-attached DAC/dongle.  Curiously I noticed that my SMSL M500 DAC display the USB input signal as DSD 22.5792MHz (see first photo below), rather than as PCM 48kHz when I would get sound from my DAC using previous build 20220904 (see second photo below).

![20230112_210018](https://user-images.githubusercontent.com/3828086/212225146-ed5a72d1-2c0a-428e-91a4-04d42c84e46a.jpg)

![20230112_202321](https://user-images.githubusercontent.com/3828086/212225605-baedcd0a-11d2-46e6-ba22-58b397185faa.jpg)

**Reply by R101G** [reply]
Why not just start again with the current image? If you use an old image, and then update, you are probably not updating the kernel, and any configuration problems are being carried over. I suggest you start from scratch with the latest image, and see if a basic configuration works.

**Reply by digixmax** [reply]
I had wanted to start with the current build, but the latest build with a binary image download link I was able to find on [the release listing](https://github.com/rern/rAudio-1/releases) is 20221117.

Do you have a binary image download link for the current build?

TIA.

**Reply by R101G** [reply]
Sorry, my mistake. I assumed there was a newer image. 

Other people are using USB DACs with no problem, so there must be something unusual in your setup. Have you tried it without DSD enabled and a reboot?

**Reply by digixmax** [reply]
> Other people are using USB DACs with no problem, so there must be something unusual in your setup. Have you tried it without DSD enabled and a reboot?

My configuration settings are pretty barebone -- Hostname, Timezone, Client ID and Secret Key for the Spotify Renderer, and CFS entries for a Shared music folder on my desktop drive -- nothing else.  The same exact rAudio settings and same USB-attached equipment chains work fine when I revert to build 2022090, so it seems that something introduced in the newer release 2022117 and/or kernel update is causing the issue.

**Comment by kw-sudo** [comment]
...I feel like not like Buddy Holly, more Sherlock Holmes. ;-)

Okay, I also did a clean install from scratch with release i20221117, nothing else, no r/Audio addons or pacman update. Left the config as clean as possible, only add my network storage.

Hard to believe, it works in any way. It doesn't matter whether the amplifier is on or off, whether the Raspberry or the amplifier is switched on first.

I only noticed one small thing: even my 'audio user' has the necessary privileges, r/Audio could not create the mount point and fstab automatically and there was an error when mounting the NAS. I had to do that manually.

Now I'm trying step-by-step what could be causing the error.

**Reply by kw-sudo** [reply]
I recap that in single steps by editing the post:

1. update addons to rAudio 20230101 --> unchanged, everything's okay
2. edit/clear config (Features/Player/Network/System), only 'Bit-Perfect', _No Volume_ is active --> unchanged
3. pacman -Syu, full system upgrade --> unchanged...

Right now, I'm speechless and at a loss. I don't know what went wrong with all the partial updates in the past. I'll keep an eye on this to see if that was just coincidence or when Murphy's law will strike again.

**Comment by digixmax** [comment]
FWIW -- in digging a bit further, I found that the device entry in spotifyd.conf gets stuck with the value "default:CARD=Headphones"  instead of getting reset to "default:CARD=M500" when my USB DAC M500 gets detected -- see two attached files rpi3 (build 20230101) vs rpi4 (build 20220904) below.

[rpi3.txt](https://github.com/rern/rAudio-1/files/10415334/rpi3.txt)
[rpi4.txt](https://github.com/rern/rAudio-1/files/10415335/rpi4.txt)

**Reply by R101G** [reply]
Does that mean you did not try disabling the on-board audio?

**Reply by digixmax** [reply]
> Does that mean you did not try disabling the on-board audio?

No, I have not yet tried disabling the on-board audio -- since I generally prefer not having to use extraneous config settings as workarounds unless I find myself desperate to get on the new build for some critical bug fixes or some new nifty features.

**Reply by R101G** [reply]
It would help diagnosis, though. It looks as if it is trying to use the mixer from one card and sending audio to the other.

**Reply by digixmax** [reply]
> It would help diagnosis, though. It looks as if it is trying to use the mixer from one card and sending audio to the other.

Ok -- disabling the onboard audio (via dtparam=audio=off in /boot/config.txt) fixes my issue of no audio signal out to USB!

Thanks.
